File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-15.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        l, r = 0, len(nums) - 1

        while l < r:
            m = (l + r) // 2
            if nums[r] > nums[m]:
                r = m
            else:
                l = m + 1
        minIndex = r
        l, r = 0, len(nums) - 1

        if target >= nums[minIndex] and target <= nums[r]:
            l = minIndex
        else:
            r = minIndex - 1

        while l <= r:
            m = (l + r) // 2
            if nums[m] < target:
                l = m + 1
            elif nums[m] > target:
                r = m - 1
            else:
                return m
        return -1

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-15.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> seen(nums.begin(), nums.end());

        int longest = 0;
        for (const auto &num : seen) {
            if (seen.find(num - 1) == seen.end()) {
                int len = 0;
                while (seen.find(num + ++len) != seen.end());
                longest = max(longest, len);
            }
        }

        return longest;
    }
};
